### PR TITLE
[Misc] Move authentication of user credentials inside PswgUserDatabas…

### PR DESCRIPTION
**Pull request by madsboddum**
…e abstraction

Relates to GitHub issue #1244

### CODE REVIEW

File: src/test/java/com/projectswg/holocore/resources/support/data/server_info/mongodb/MongoDBTestContainer.kt [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+ * along with Holocore.  If not, see <http://www.gnu.org/licenses/>.               *
+ ***********************************************************************************/
+package com.projectswg.holocore.resources.support.data.server_info.mongodb
+
+import com.mongodb.client.MongoClient
+import com.mongodb.client.MongoClients
+import org.testcontainers.containers.MongoDBContainer
+import org.testcontainers.utility.DockerImageName
+
+class MongoDBTestContainer {
```
Comment (Josh-Larson) [inline src/test/java/com/projectswg/holocore/resources/support/data/server_info/mongodb/MongoDBTestContainer.kt:34]: You should make this an `object` and use the shutdown hooks to cleanup

File: src/test/java/com/projectswg/holocore/resources/support/data/server_info/mongodb/MongoDBTestContainer.kt [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+	private val mongoDBContainer = MongoDBContainer(DockerImageName.parse("mongo:5.0.14"))
+
+	fun start(): MongoClient {
+		mongoDBContainer.start()
```
Comment (Josh-Larson) [inline src/test/java/com/projectswg/holocore/resources/support/data/server_info/mongodb/MongoDBTestContainer.kt:38]: Then you can create a public variable `client` that's lazily initialized:
```kt
val client by lazy {
    val container = MongoDBContainer(DockerImageName.parse("mongo:5.0.14"))
    container.start()
    MongoClients.create(container.connectionString)
}

Comment (Josh-Larson) [reply src/test/java/com/projectswg/holocore/resources/support/data/server_info/mongodb/MongoDBTestContainer.kt:38]: Shutdown hook might look something like this
```kt
Runtime.getRuntime().addShutdownHook(Thread { container.stop() })
```

File: src/test/java/com/projectswg/holocore/resources/support/data/server_info/mongodb/MongoDBTestContainer.kt (hunk 1 above)
Comment (madsboddum) [reply src/test/java/com/projectswg/holocore/resources/support/data/server_info/mongodb/MongoDBTestContainer.kt:34]: Oh yeah, forgot about that. Will fix.

**Pull request merged**